numpy/core/tests/test_scalar_methods.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test the scalar constructors, which also do type-coercion
+"""
+from __future__ import division, absolute_import, print_function
+
+import fractions
+import numpy as np
+
+from numpy.testing import (
+    run_module_suite,
+    assert_equal, assert_almost_equal, assert_raises, assert_warns,
+    dec
+)
+
+float_types = [np.half, np.single, np.double, np.longdouble]
+
+def test_float_as_integer_ratio():
+    # derived from the cpython test "test_floatasratio"
+    for ftype in float_types:
+        for f, ratio in [
+                (0.875, (7, 8)),
+                (-0.875, (-7, 8)),
+                (0.0, (0, 1)),
+                (11.5, (23, 2)),
+            ]:
+            assert_equal(ftype(f).as_integer_ratio(), ratio)
+
+        rstate = np.random.RandomState(0)
+        fi = np.finfo(ftype)
+        for i in range(1000):
```

Comment:
I don't see the point of repeating the same 1000 numbers every time, especially with a `try/except` - I think it is much better to use actual examples, say `np.pi`, `sqrt(2)`, `1/9`, including ones where the `try/except` would fail (presumably very large or small exponent).

Reply:
Sounds like a good fit for [`hypothesis`](https://github.com/HypothesisWorks/hypothesis) to generate some cases for us.

Reply:
I've replaced the 1000 iterations with a small set of cases per given dtype generated using `hypothesis`, using the draft code on this gist: https://gist.github.com/tylerjereddy/35936be43743527c16246837295161fe

Is that a reasonable compromise?

Reply:
Looks fine to me. At this point, this code has been well tested by virtue of going through a bunch of different versions of the test anyway!